demos/heat_equation/setup.py

```diff
@@ -2,11 +2,11 @@
 from firedrake_adjoint import * 
 
 
-def initial_mesh():
+def initial_mesh(n):
     """
     Construction of initial mesh
     """
-    return RectangleMesh(5, 5, (-1,1), (-1,1))
+    return RectangleMesh(5*n, 5*n, (-n,n), (-n,n))
```

Comment:
Also I think it would be better to have `RectangleMesh(5 * 2 ** n, 5 * 2 ** n, (-1, 1), (-1, 1))` as I recommended before because then increasing `n` by one is like doing uniform refinement.